### PR TITLE
CDAP-4075 Added checks for handling pre 3.4 Workflow applications.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-4075

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT3759-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java [hunk 1]
```diff
@@ -198,7 +199,9 @@ private Workflow initializeWorkflow() throws Exception {
     transactionContext.start();
 
     try {
-      workflow.initialize(basicWorkflowContext);
+      if (workflow instanceof ProgramLifecycle) {
+        ((ProgramLifecycle) workflow).initialize(basicWorkflowContext);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: Should cast to `ProgramLifecycle<WorkflowContext>`.

**Comment by chtyim** [comment]
Just one minor comment. Rest LGTM.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java]: Fixed. Done same for the destroy method as well.

**Comment by sagarkapare** [comment]
Thanks @chtyim for the review. Squashed the commit and started new build here - http://builds.cask.co/browse/CDAP-DUT3759-2

**Comment by sagarkapare** [comment]
Build is green. Merging.

**Pull request merged**